server/backend/server.js

```diff
@@ -75,34 +73,33 @@ var findRoad = (parsed, db, callback) => {
     }
   };
 
-  segments.findOne(q, {speed: 1, road_id: 1, _id: 0}, (err, segment) => {
-    if (err) {
-      // Collection read error
-      console.log(err);
-      resp.online = 0;
-      callback(resp);
-    } else if (!segment) {
-      // No segment match!
-      callback(resp);
+  const resp = newResponse();
+  const segments = db.collection(config.segments);
+  const findOneSegment = denodeifyMethod(segments, 'findOne');
+
+  return findOneSegment(q, {speed: 1, road_id: 1, _id: 0})
+  .then({road_id, speed} => {
+    if (!segment) {
+      return resp;
```

Comment:
Returning anything from a `thenable` will wrap it in a `Promise`, this enables chaining and error propagation to the consumer.

This return statement is called when `findOneSegment` executes successfully but does not find a segment.